test/e2e-tests/tests/pipelines_test.go

```diff
@@ -0,0 +1,156 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/opendatahub-io/ai-edge/test/e2e-tests/support"
+	"golang.org/x/net/context"
+	k8sApi "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PipelineRelativePath    = "../../../pipelines/tekton/aiedge-e2e"
+	PipelineRunRelativePath = "../../../pipelines/tekton/aiedge-e2e/aiedge-e2e.pipelinerun.yaml"
+	S3BucketNameEnvKey      = "S3_BUCKET"
+	RegistryRepoNameEnvKey  = "REGISTRY_REPO"
+)
+
+func CreateContext() context.Context {
+	return context.Background()
+}
+
+func WaitForAllPipelineRunsToComplete(ctx context.Context, clients *support.Clients) error {
+CheckRuns:
+	for {
+		list, err := clients.PipelineRun.List(ctx, k8sApi.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, pipelineRun := range list.Items {
+			if len(pipelineRun.Status.Conditions) == 0 {
+				time.Sleep(time.Second * 5)
+				continue CheckRuns // waiting for conditions to exist first
+			}
+
+			for _, condition := range pipelineRun.Status.Conditions {
+				if condition.Reason == "Failed" {
+					return fmt.Errorf("pipelineRun \"%v\" failed with message \"%v\"", pipelineRun.Name, condition.Message)
+
+				} else if condition.Reason == "Running" {
+					time.Sleep(time.Second * 5)
+					continue CheckRuns // it is still running so restart the loop and wait 5 seconds
+				}
+			}
+		}
+
+		break // none failed and none were still running so all pipeline runs have complete
+	}
+
+	return nil
+}
+
+// init is called before any test. This is used to build
+// then apply the kustomize config for the main pipeline
+func init() {
+
+	resourceMap, err := support.ApplyPipelineKustomizeFolder(PipelineRelativePath)
+	if err != nil {
+		panic(fmt.Sprintf("error while building kustomize : %v", err.Error()))
+	}
+
+	ctx := CreateContext()
+
+	clients, err := support.CreateClients("test-namespace")
+	if err != nil {
+		panic(fmt.Sprintf("error while creating client : %v", err.Error()))
+	}
+
+	for _, resource := range resourceMap.Resources() {
+		err = support.CreateObjectFromResource(ctx, &clients, resource)
+		if err != nil {
+			panic(fmt.Errorf("error while creating objects from kustomize resources : %v", err.Error()))
+		}
+	}
+}
+
+func Test_PipelineExists(t *testing.T) {
+	t.Skip()
+	ctx := CreateContext()
+
+	clients, err := support.CreateClients("test-namespace")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	list, err := clients.Pipeline.List(ctx, k8sApi.ListOptions{})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	expectedPipelineNames := []string{"aiedge-e2e"}
+	if len(list.Items) != len(expectedPipelineNames) {
+		t.Fatalf("expected %v pipelines but %v existed on the cluster", len(expectedPipelineNames), len(list.Items))
+	}
+
+OUTER:
+	for _, name := range expectedPipelineNames {
+		for _, pipeline := range list.Items {
+			if pipeline.Name == name {
+				continue OUTER
+			}
+		}
```

Comment:
This can be extracted to a function which would return true if the pipeline was found and one if, which would check the outcome. This way the label can be omitted. But again, it may be just me writing it differently, no strong opinion :)

Reply:
Deleted